src/components/signInForm.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <div class="wrap">
+    <h1>Sign In</h1>
+    <p>
+      Don’t have an account, please
+      <router-link to="/signup">Sign Up</router-link>
+    </p>
+  <form v-on:submit.prevent="">
+    <input
+      type="text"
+      placeholder="Email"
+      v-model="userEmail"
+    >
+    <input
+      type="password"
+      placeholder="Password"
+      v-model="userPassword"
+    >
+    <button
+      class="sign-in"
+      @click="signIn"
+    >Sign In</button>
+  </form>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'signInForm',
+  data: () => ({
+    userEmail: '',
+    userPassword: '',
+  }),
+  methods: {
+    signIn() {
+      const emptyStr = /(^\s{1,}$)|(^.{0}$)/;
```

Comment:
Most probably you dont need to use regExp for checking empty string but just, just **trim** it and check like Boolean value